src/V3Assert.cpp

```diff
@@ -571,11 +583,281 @@ class AssertVisitor final : public VNVisitor {
     }
 };
 
+class AssertCtlVisitor final : public VNVisitor {
```

Comment:
Please put comments at top of file as to how the algorithm works - see other files for examples.